### PR TITLE
UIIN-2670:Jest/RTL: Cover LocationSelectionWithCheck components with unit tests

### PR DESCRIPTION
**Pull request by UladzislauKutarkin**
https://folio-org.atlassian.net/browse/UIIN-2670
Jest/RTL: Cover LocationSelectionWithCheck components with unit tests.
image

### CODE REVIEW

File: src/edit/common/LocationSelectionWithCheck.test.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  Check: {
+    useByLocation: jest.fn(() => jest.fn(() => true)), // Mocking always true for check
+  },
+  Confirmation: {
+    Heading: jest.fn(() => 'Remote Storage Heading'),
+    Message: jest.fn(() => 'Remote Storage Message'),
+  },
+}));
+
+jest.mock('@folio/stripes/smart-components', () => ({ ...jest.requireActual('@folio/stripes/smart-components'),
```
Comment (BogdanDenis) [inline src/edit/common/LocationSelectionWithCheck.test.js:22]: Please format this block so:
1. each item is on a new line
2. there is a trailing coma after each item
3. each jsx prop is on a new line

**Review by BogdanDenis — CHANGES_REQUESTED** [review]

File: src/edit/common/LocationSelectionWithCheck.test.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    renderWithIntl(renderWithRouter(renderWithFinalForm(
+      <LocationSelectionWithCheck name="location" input={inputMock} {...restMock} />
+    )));
+
+    const confirmButton = screen.getByRole('button', { name: 'confirm' });
+
+    act(() => fireEvent.click(screen.getByTestId('selection-id')));
+
+    fireEvent.click(confirmButton);
+  });
```
Comment (Dmytro-Melnyshyn) [inline src/edit/common/LocationSelectionWithCheck.test.js:67]: Please add `expect` to both tests.

File: src/edit/common/LocationSelectionWithCheck.test.js (hunk 1 above)
Comment (UladzislauKutarkin) [reply src/edit/common/LocationSelectionWithCheck.test.js:22]: Done!

**Review by BogdanDenis — APPROVED** [review]

**Review by vashjs — APPROVED** [review]

File: src/edit/common/LocationSelectionWithCheck.test.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    renderWithIntl(renderWithRouter(renderWithFinalForm(
+      <LocationSelectionWithCheck name="location" input={inputMock} {...restMock} />
+    )));
+
+    const cancelButton = screen.getByRole('button', { name: 'cancel' });
+
+    act(() => fireEvent.click(screen.getByTestId('selection-id')));
+
+    fireEvent.click(cancelButton);
+    expect(inputMock.onChange).toHaveBeenCalled();
```
Comment (Dmytro-Melnyshyn) [inline src/edit/common/LocationSelectionWithCheck.test.js:91]: Please clear this mock.

**Review by Dmytro-Melnyshyn — CHANGES_REQUESTED** [review]

File: src/edit/common/LocationSelectionWithCheck.test.js (hunk 3 above)
Comment (UladzislauKutarkin) [reply src/edit/common/LocationSelectionWithCheck.test.js]: done

**Pull request merged**